### PR TITLE
C++ front-end: use ranged for

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/cpp/cpp_typecheck_fargs.cpp [hunk 1]
```diff
@@ -40,8 +38,8 @@ void cpp_typecheck_fargst::build(
   operands.clear();
   operands.reserve(function_call.op1().operands().size());
 
-  for(std::size_t i=0; i<function_call.op1().operands().size(); i++)
-    operands.push_back(function_call.op1().operands()[i]);
+  for(const auto &op : function_call.op1().operands())
+    operands.push_back(op);
```
Comment (tautschnig) [inline src/cpp/cpp_typecheck_fargs.cpp]: Rewrite this (and some lines above) as `operands = function_call.op1().operands();` ?

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 7137d0606450e4388b70f1c5e937183e1c9894aa).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86369132
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

File: src/cpp/cpp_typecheck_resolve.cpp [hunk 2]
```diff
@@ -483,7 +462,7 @@ void cpp_typecheck_resolvet::disambiguate_functions(
         1000*template_distance+args_distance;
 
       distance_map.insert(
-        std::pair<std::size_t, exprt>(total_distance, *it));
+        std::pair<std::size_t, exprt>(total_distance, old_id));
```
Comment (tautschnig) [inline src/cpp/cpp_typecheck_resolve.cpp]: More modernisation: `distance_map.insert({total_distance, old_id});`

File: src/cpp/cpp_constructor.cpp [hunk 3]
```diff
@@ -234,11 +228,8 @@ optionalt<codet> cpp_typecheckt::cpp_constructor(
       cpp_namet(constructor_name, source_location).as_expr();
     function_call.arguments().reserve(operands_tc.size());
 
-    for(exprt::operandst::iterator
-        it=operands_tc.begin();
-        it!=operands_tc.end();
-        it++)
-      function_call.op1().copy_to_operands(*it);
+    for(auto &op : operands_tc)
+      function_call.op1().copy_to_operands(op);
```
Comment (tautschnig) [inline src/cpp/cpp_constructor.cpp]: I think we can get rid of this loop entirely if using a 3-argument constructor for `function_call` above.

File: src/cpp/cpp_id.cpp [hunk 4]
```diff
@@ -63,22 +61,16 @@ void cpp_idt::print_fields(std::ostream &out, unsigned indent) const
...
 
-  for(scope_listt::const_iterator
-      it=secondary_scopes.begin();
-      it!=secondary_scopes.end();
-      it++)
+  for(const auto &s : secondary_scopes)
   {
     for(unsigned i=0; i<indent; i++) out << ' ';
-    out << "  secondary_scope=" << (*it)->identifier << '\n';
+    out << "  secondary_scope=" << s->identifier << '\n';
```
Comment (tautschnig) [inline src/cpp/cpp_id.cpp:67]: We could further simplify this loop by using `std::string(indent, ' ')` in place of the `for` loop above. Applies several times across this file.

**Review by tautschnig — APPROVED** [review]
Great, with even more room for related cleanup I believe.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 02b517fd5c5c7bf6ff1bee6d5cd7754498a9f24b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86369462

**Comment by kroening** [comment]
All done!

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 76f145ddcf2470a0fa704518d9027ed60429a063).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86389696

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 4bae1df6ae3328398416e987492a5203de2c5fc0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86391508

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**